File: Gherkin/Token.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Gherkin
{
    public class Token
    {
        public bool IsEOF { get { return Line == null; } }
        public IGherkinLine Line { get; set; }
        public TokenType MatchedType { get; set; }
        public string MatchedKeyword { get; set; }
        public string MatchedText { get; set; }
        public GherkinLineSpan[] MathcedItems { get; set; }
        public int MatchedIndent { get; set; }
        public GherkinDialect MatchedGherkinDialect { get; set; }

        public Token(IGherkinLine line)
        {
            Line = line;
        }

        public void Detach()
        {
            Line.Detach();
        }

        public override string ToString()
        {
            return string.Format("{0}: {1}/{2}", MatchedType, MatchedKeyword, MatchedText);
        }
    }
}

File: swift/xcode_proj/gherkin/gherkin/Token.swift
import Foundation

struct Token {
    var line: GherkinLine
    var location: [String: String]
    var matchedType: TokenType?
    var matchedText: String?
    var matchedKeyword: [String]?
    var matchedIndent: Int?
    var matchedItems: [String]?
    var matchedGherkinDialect: String?
    
    init(line: GherkinLine, location: [String: String]) {
        self.line = line
        self.location = location
    }
    
    func eof() -> Bool {
        self.line
    }
    
}
